magnum/solvers/search_oracle.py

```diff
@@ -0,0 +1,78 @@
+from itertools import cycle
+
+import stl
+import numpy as np
+
+import traces
+from functools import reduce
+
+def random_series(g):
+    times = (g.model.dt*t for t in g.times)
+    return traces.TimeSeries(zip(times, np.random.random(len(g.times))))
+
+def extract_t(controls, w_star, t):
+    controls_matrix = np.array([np.array(control_i).T[1] for control_i
+                                    in controls.values()])
+    w_matrix = np.array([np.array(w_i).T[1] for w_i in w_star.values()])
+
+    return {'u':controls_matrix[:, [t]], 'w':w_matrix.T[:,[t]]}
+
+def check_sat(w_star, u_star, r, g, num_samples=100):
+    A, B, C = g.model.dyn
+    dt = g.model.dt
+    # Adjust for discrete time
+    A = np.eye(len(g.model.vars.state)) + dt * A
+    B = dt * B
+    C = dt * C
+
+    times = (g.model.dt * t for t in g.times)
```

Comment:
I justed added some code in `game.py` for scaled times. 

`times = g.scaled_times`